Meta/CMake/flatpak/org.ladybird.Ladybird.json

```diff
@@ -82,18 +82,141 @@
         "/bin"
       ]
     },
+    {
+      "name": "gn",
+      "buildsystem": "simple",
+      "sources": [
+        {
+          "type": "git",
+          "url": "https://gn.googlesource.com/gn",
+          "commit": "97b68a0bb62b7528bc3491c7949d6804223c2b82"
+        },
+        {
+          "type": "file",
+          "path": "gn-whole-enchilada.sh"
+        }
+      ],
+      "build-commands": [
+        "./gn-whole-enchilada.sh"
+      ],
+      "cleanup": [ "*" ]
+    },
     {
       "name": "angle",
       "buildsystem": "simple",
-      "build-options": {
-        "build-args": [
-          "--share=network"
-        ]
-      },
+      "build-commands": [
+        "./angle-checkout.sh",
+        "./angle-configure.sh",
+        "./angle-build.sh",
+        "./angle-install.sh"
+      ],
       "sources": [
```

Comment:
the final list of sources you have here looks... close? I don't think we want to download all these sysroots though.

It would be nice to avoid the gclient post-hooks as well. Or at least, avoid the rather exciting list of files that are added to our repo to support calling those python scripts.

I'd rather patch the gn build to skip things that we aren't using than download a bunch of extra stuff to fit into the 'standard chromium build flow' or w/e.

Reply:
Thanks for this helpful background info!

Do you see value in moving on with this PR?
The list of dependencies might make it easier to strip down the build, compared to the current state.
There is still some cleanup that has to be done before the actual merge.

Reply:
> so trying to integrate with the existing gn build seems prudent.

As a side step, I'm looking into the gn build on Linux (see `#build-problems`). Have to see what remains after the directory structure is lined up again. If I succeed, it will be a separate PR (and maybe a prerequisite for this one?).

Reply:
> fixup-squashed with all the angle-specific files in an angle subfolder of Meta/CMake/flatpak would probably be enough to get us a workable offline build

Done. I will close this thread and get the PR out of draft, to have the finishing off in the review.

Reply:
oh, hold up, I did that work already :sweat_smile:

Reply:
ah, you did a squash. hm. I have a branch with several commits chunked up

Reply:
nah, i've been working in tandem and forgot to comment in the thread here